### ISSUE TITLE
Ignorance of Labeled Training Samples Selection？

### ISSUE THREAD

**Issue by dsh65** [vector-1127/DAC#21]
@vector-1127 i only find the section in cifar-10_2
```
def my_binary_crossentropy(y_true, y_pred):
            w1 = K.switch(y_true<lower,1,0)
            w2 = K.switch(y_true>upper,1,0)
            w = w1 + w2
            nb_selected = K.sum(w)
            y_true = K.switch(y_true>(upper+lower)/2,1,0)
            return K.sum(w*K.binary_crossentropy(y_true, y_pred), axis=-1)/nb_selected
model.compile(optimizer=RMSprop(0.001), 
              loss={'main_output':my_binary_crossentropy,'aux_output':'binary_crossentropy'}, 
              loss_weights={'main_output':1,'aux_output':1})
for i in range(X_train.shape[0]//nb):
                Xbatch = X_train[index[np.arange(i*nb,(i+1)*nb)]]
                Y = cluster_l2.predict(Xbatch)
                Y_l1 = cluster_l1.predict(Xbatch)
                Ybatch = np.dot(Y,Y.T)
                for k in range(nb_epoch):
                    np.random.shuffle(index_loc)
                    for j in range(Xbatch.shape[0]//batch_size):
                        address = index_loc[np.arange(j*batch_size,(j+1)*batch_size)]
                        X_batch = Xbatch[address]
                        Y_batch = Ybatch[address,:][:,address]
                        Y_batch_l1 = Y_l1[address]
                        sign = 0
                        for X_batch_i in datagen.flow(X_batch, batch_size=batch_size,shuffle=False):
                            loss = model.train_on_batch([X_batch_i],[Y_batch_l1, Y_batch])
                            sign += 1
                            if sign>1:
                                break
```
but in the other codes, i find it is replaced by
```
for i in range(X_train.shape[0]//nb):
                Xbatch = X_train[index[np.arange(i*nb,(i+1)*nb)]]
                Y = cluster_l2.predict(Xbatch)
                **Ybatch = (np.sign(np.dot(Y,Y.T)-th)+1)/2**
                for k in range(nb_epoch):
                    np.random.shuffle(index_loc)
                    for j in range(Xbatch.shape[0]//batch_size):
                        address = index_loc[np.arange(j*batch_size,(j+1)*batch_size)]
                        X_batch = Xbatch[address]
                        Y_batch = Ybatch[address,:][:,address]
                        Y_ = Y[address]
                        sign = 0
                        for X_batch_i in datagen.flow(X_batch, batch_size=batch_size,shuffle=False):
                            loss = model.train_on_batch([X_batch_i],[Y_, Y_batch])
                            sign += 1
                            if sign>1:
                                break
model.compile(optimizer=RMSprop(0.001), 
              loss={'main_output':'binary_crossentropy','aux_output':'binary_crossentropy'}, 
              loss_weights={'main_output':1,'aux_output':10})
```
is there a simpilification about Labeled Training Samples Selection? pairs of pics with similarities between upper and lower are seems used to calculate the loss.
